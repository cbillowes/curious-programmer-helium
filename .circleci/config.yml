version: 2
jobs:
  build:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Chopping onions without the tears"
      - run: npm run build
  test:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making soup and drinking beers"
      - run: sudo npm install -g jest-cli
      - run: npm run test
  page_speed:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making sure it won't take years"
      - run: echo "* Deploy to Netlify"
      - run: echo "* Execute lighthouse page speed tests on build"
workflows:
  version: 2
  integration:
    jobs:
      - build
      - test:
          requires:
            - build
      - page_speed:
          requires:
            - test
