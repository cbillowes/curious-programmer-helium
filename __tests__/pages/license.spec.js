import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Helmet from "react-helmet"
import LicensePage from "../../src/pages/license"
import config from "../../data/SiteConfig"

describe("a license page", () => {
  it("should render", () => {
    const component = <LicensePage />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should generate the meta data title", () => {
    const component = mount(<LicensePage />)
    const helmet = Helmet.peek()
    expect(helmet.title).toEqual(`License | ${config.siteTitle}`)
  })

  it("should render children tags", () => {
    const component = mount(<LicensePage />)
    expect(component.find(".license-container").html()).toBeDefined()
  })
})