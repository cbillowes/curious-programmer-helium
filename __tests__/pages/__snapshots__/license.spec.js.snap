// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a license page should render 1`] = `
<div>
  <div
    className="navigation"
  >
    <div
      className="logo"
    >
      <div
        className="default-logo"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          { 
          <span
            className="normal-weight"
          >
            curious
          </span>
           
          programmer
           }
        </a>
      </div>
      <div
        className="mobile-logo"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          { }
        </a>
      </div>
    </div>
    <div
      className="icons"
    >
      <div
        className="icon search "
        onClick={[Function]}
      >
        <i />
      </div>
      <div
        className="icon menu "
        onClick={[Function]}
      >
        <i />
      </div>
    </div>
    <nav
      className="menubar "
      onClick={[Function]}
    >
      <ul>
        <li
          className="menu-item selected"
        >
          <a
            aria-current="page"
            className=""
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={Object {}}
          >
            Blog
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/tags"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Tags
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/credits"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Credits
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/privacy-policy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/license"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            License
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="searchbar "
    >
      <input
        className="search-query"
        onKeyUp={[Function]}
        type="text"
      />
      <div
        className="results"
      >
        <ul>
          <li
            className="empty"
          >
            What do you feel like reading today? Search as you type, also, enter works too.
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="license-container container"
  >
    <h1>
      License
    </h1>
    <section>
      <h2>
        The Code
      </h2>
      <p
        className="emphasize"
      >
        <strong>
          Applies to
        </strong>
         all code written explicitly by Clarice Bouwer. It does not include third-party libraries nor code snippets exempt from this license. Such snippets will be explicitly referenced in a link to this document.
      </p>
      <h3>
        The MIT License, unless otherwise stated
      </h3>
      <p>
        Copyright © 
        <time
          dateTime={"2019-01-05T06:58:00.751Z"}
        >
          2019
        </time>
         Curious Programmer
      </p>
      <p>
        Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
      </p>
      <p>
        The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
      </p>
      <p>
        THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
      </p>
    </section>
    <section>
      <h2>
        The Content
      </h2>
      <h3>
        Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
      </h3>
      <span
        className="license"
      >
        <a
          className="license-icon"
          href="http://creativecommons.org/licenses/by-sa/4.0/"
          rel="nofollow noopener noreferrer license"
          target="_blank"
        >
          <img
            alt="Creative Commons License"
            src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
            title="Creative Commons Attribution-ShareAlike 4.0 International License"
          />
        </a>
        <p>
          <span
            property="dct:title"
            xmlns:dct="http://purl.org/dc/terms/"
          >
             
            Curious Programmer
             content
             
          </span>
           
          by
           
          <a
            className=""
            href="https://curiousprogrammer.io"
            rel="nofollow noopener noreferrer cc:attributionURL"
            target="_blank"
          >
            Clarice Bouwer
          </a>
           
          is licensed under a
           
          <a
            className=""
            href="http://creativecommons.org/licenses/by-sa/4.0/"
            rel="nofollow noopener noreferrer license"
            target="_blank"
          >
            Creative Commons Attribution-ShareAlike 4.0 International License
          </a>
          .
        </p>
      </span>
      You are free to:
      <ul>
        <li>
          <strong>
            Share
          </strong>
           — copy and redistribute the material in any medium or format
        </li>
        <li>
          <strong>
            Adapt
          </strong>
           — remix, transform, and build upon the material for any purpose, even commercially.
        </li>
      </ul>
      Curious Programmer cannot revoke these freedoms as long as you follow the license terms:
      <ul>
        <li>
          <strong>
            Attribution
          </strong>
           — You must give appropriate credit, provide a link to the license, and indicate if changes were made. You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
        </li>
        <li>
          <strong>
            ShareAlike
          </strong>
           — If you remix, transform, or build upon the material, you must distribute your contributions under the same license as the original.
        </li>
      </ul>
      <p>
        You may not apply legal terms or technological measures that legally restrict others from doing anything the license permits.
      </p>
      <p
        className="emphasize"
      >
        This is just a taste. You'll probably need to read the 
         
        <a
          className=""
          href="https://creativecommons.org/licenses/by-sa/4.0/legalcode"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          full license
        </a>
        .
      </p>
    </section>
    <section>
      <h2>
        Photography
      </h2>
      <p
        className="emphasize"
      >
        I use 
        <a
          className=""
          href="https://picsum.photos/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Lorem Picsum
        </a>
         
        which provides backgrounds by
         
        <a
          className=""
          href="https://unsplash.com/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Unsplash
        </a>
        .
         
        The 
        <a
          className=""
          href="https://unsplash.com/license"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          License
        </a>
         states that all photos can be used for free.
      </p>
      <p>
        All photos published on Unsplash can be used for free. You can use them for commercial and noncommercial purposes. You do not need to ask permission from or provide credit to the photographer or Unsplash, although it is appreciated when possible.
      </p>
      <p>
        More precisely, Unsplash grants you an irrevocable, nonexclusive, worldwide copyright license to download, copy, modify, distribute, perform, and use photos from Unsplash \\ for free, including for commercial purposes, without permission from or attributing the photographer or Unsplash. This license does not include the right to compile photos from Unsplash to replicate a similar or competing service.
      </p>
    </section>
  </div>
  <div
    className="footer-container not-for-print"
  >
    <footer
      className="footer"
    >
      <ul
        className="social-links-footer"
      >
        <li
          className="link"
        >
          <a
            className="github"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
        <li
          className="link"
        >
          <a
            className="twitter"
            href="https://twitter.com/cbillowes"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
        <li
          className="link"
        >
          <a
            className="linkedin"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
      </ul>
      <div
        className="copyright-footer"
      >
        Copyright © 
        
        2019
         
        
        Curious Programmer
        . 
        
        Made with Gatsby v2 and other 
        
        <a
          href="/credits"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          cool stuff
        </a>
        .
      </div>
      <div
        className="license-footer"
      >
        <span
          className="license"
        >
          <a
            className="license-icon"
            href="http://creativecommons.org/licenses/by-sa/4.0/"
            rel="nofollow noopener noreferrer license"
            target="_blank"
          >
            <img
              alt="Creative Commons License"
              src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
              title="Creative Commons Attribution-ShareAlike 4.0 International License"
            />
          </a>
          <p>
            <span
              property="dct:title"
              xmlns:dct="http://purl.org/dc/terms/"
            >
               
              Curious Programmer
               content
               
            </span>
             
            by
             
            <a
              className=""
              href="https://curiousprogrammer.io"
              rel="nofollow noopener noreferrer cc:attributionURL"
              target="_blank"
            >
              Clarice Bouwer
            </a>
             
            is licensed under a
             
            <a
              className=""
              href="http://creativecommons.org/licenses/by-sa/4.0/"
              rel="nofollow noopener noreferrer license"
              target="_blank"
            >
              Creative Commons Attribution-ShareAlike 4.0 International License
            </a>
            .
          </p>
        </span>
         
         
        <p>
          Also, there is a 
           
          <a
            href="/license"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            License
          </a>
           for the code 
          and it's probably a good idea to read the
           
          <a
            href="/privacy-policy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy Policy
          </a>
          .
        </p>
      </div>
    </footer>
  </div>
</div>
`;
